xarray/core/dataarray.py

```diff
@@ -788,37 +788,47 @@ def reindex(self, method=None, tolerance=None, copy=True, **indexers):
             method=method, tolerance=tolerance, copy=copy, **indexers)
         return self._from_temp_dataset(ds)
 
-    def rename(self, new_name_or_name_dict):
+    def rename(self, new_name_or_dims_dict, new_dims_dict=None):
         """Returns a new DataArray with renamed coordinates and/or a new name.
 
 
         Parameters
         ----------
-        new_name_or_name_dict : str or dict-like
+        new_name_or_dims_dict : str or dict-like
             If the argument is dict-like, it it used as a mapping from old
-            names to new names for coordinates (and/or this array itself).
-            Otherwise, use the argument as the new name for this array.
+            names to new names for coordinates. Otherwise, use the argument
+            as the new name for this array.
+
+        new_dims_dict : dict-like
```

Comment:
Instead of adding a second argument here, I would simply require calling `.rename()` twice if you want to rename both the DataArray and it's coordinates at the same time.